src/Microsoft.FeatureManagement/FeatureManagementBuilderExtensions.cs

```diff
@@ -39,6 +40,41 @@ public static class FeatureManagementBuilderExtensions
             return builder;
         }
 
+        /// <summary>
+        /// Adds a <see cref="VariantServiceProvider{TService}"/> to the feature management system.
+        /// </summary>
+        /// <param name="builder">The <see cref="IFeatureManagementBuilder"/> used to customize feature management functionality.</param>
+        /// <param name="featureName">The variant feature flag used to assign variants. The <see cref="VariantServiceProvider{TService}"/> will return different implementations of TService according to the assigned variant.</param>
+        /// <returns>A <see cref="IFeatureManagementBuilder"/> that can be used to customize feature management functionality.</returns>
+        public static IFeatureManagementBuilder AddVariantServiceProvider<TService>(this IFeatureManagementBuilder builder, string featureName) where TService : class
```

Comment:
```suggestion
        public static IFeatureManagementBuilder RegisterVariantService<TService>(this IFeatureManagementBuilder builder, string featureName) where TService : class
```

Reply:
I suggest the name because I think it breaks a way from typical `Add`. Here we require a parameter which is the feature name that is mapped to the variant service.

Reply:
I also wouldn't mind `WithVariantService` similar to `WithTargeting`

Reply:
I like `WithVariantService` which makes our API's method name style more consistent.